File: Pset6/vigenere.py
#   vigenere.py
#   Aamel Unia
#   aameluni@buffalo.edu
#   Basic implementation of the Vigenere Cipher.

import sys

def main():
    if len(sys.argv)!=2:
        print("Usage: python vigenere.py k")
        exit(1)
        
    key = sys.argv[1]
    
    for i in range(len(key)):
        if not key[i].isalpha():
            print("Usage: python vigenere.py k")
            exit(1)
            
    print("plaintext: ",end="")
    plaintext = input()
    
    ciphertext=""
    counter=0
    
    for c in plaintext:
        if c.isalpha():
            index = counter%len(key)
            if c.islower():
                temp = ( (ord(key[index]) - ord('a')) + (ord(c) - ord('a')) )%26+97 if key[index].islower() else ( (ord(key[index])-ord('A')) + (ord(c)-ord('a')) )%26+97
                ciphertext+=chr(temp)
            elif c.isupper():
                temp = ( (ord(key[index]) - ord('a')) + (ord(c) - ord('A')) )%26+65 if key[index].islower() else ( (ord(key[index])-ord('A')) + (ord(c)-ord('A')) )%26+65
                ciphertext+=chr(temp)
            counter+=1
        else:
            ciphertext+=c
            
    print(ciphertext)
    
    
if __name__ == "__main__":
    main()
    exit(0)

File: Pset2/vigenere.c
/*
 * vigenere.c
 * Aamel Unia
 * aameluni@buffalo.edu
 * Basic implementation of the Vigenere Cipher.
 */


#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main (int argc, string argv[]){
    
if (argc!=2) {
    printf("Incorrect number of arguments!\n"); 
    return 1;
}

string key = argv[1];
int keylen = strlen(key);

for (int i=0; i<keylen; i++){
    if(!isalpha(key[i])) {
        printf("Incorrect argument!\n");
        return 1;
    }
}

string text = GetString();


for (int i=0, counter=0, n=strlen(text); i<n; i++){
    
    if(isalpha(text[i])){
        
        if(islower(text[i])){
            
             if(islower(key[counter%keylen])){
                char temp = (key[counter%keylen]-'a' + text[i]-'a')%26 + 97; 
                text[i]=temp;
             }
             if(isupper(key[counter%keylen])){
                char temp = (key[counter%keylen]-'A' + text[i]-'a')%26 + 97; 
                text[i]=temp;
             }
             
         }
    
        else if(isupper(text[i])){
            
            if(islower(key[counter%keylen])){
                char temp = (key[counter%keylen]-'a' + text[i]-'A')%26 + 65; 
                text[i]=temp;
             }
            if(isupper(key[counter%keylen])){
                char temp = (key[counter%keylen]-'A' + text[i]-'A')%26 + 65; 
                text[i]=temp;
             }
        }
        
        counter++;
    }
}

printf("%s\n", text);

return 0;
}
